src/lib.rs

```diff
@@ -207,6 +209,23 @@ pub fn run(path: &PathBuf) -> Vec<Message> {
                             }
                         }
                     }
+
+                    if h.compressed_header && h.time_offset.is_some() {
+                        let time_offset = h.time_offset.unwrap();
+                        let date: u32 = if time_offset as u32 >= (last_timestamp & 0x1F) {
```

Comment:
It might be nice to put the `0x1F` and `0x20` values up at the top as consts, just so it's a little clearer what's happening here.